### PR TITLE
DebugInterface: Check visited addresses in GetThreads

### PR DESCRIPTION
**Pull request by sepalani**
This PR is an attempt to fix https://bugs.dolphin-emu.org/issues/12410.

Ready to be reviewed & tested.

### CODE REVIEW

File: Source/Core/Core/Debugger/PPCDebugInterface.cpp [hunk 1]
```diff
@@ -181,9 +183,13 @@ Common::Debug::Threads PPCDebugInterface::GetThreads() const
   if (!active_thread->IsValid())
     return threads;
 
-  const auto insert_threads = [&threads](u32 addr, auto get_next_addr) {
+  std::vector<u32> visited_addr{{active_thread->GetAddress()}};
```
Comment (iwubcode) [inline Source/Core/Core/Debugger/PPCDebugInterface.cpp:186]: Would a set or another container be a better choice?

Comment (AdmiralCurtiss) [reply Source/Core/Core/Debugger/PPCDebugInterface.cpp:186]: `vector<>` tends to perform really well for small sizes on modern systems thanks to cache locality, and I can't imagine the amount of addresses here reaching even three digits. So I can't imagine this mattering much.

**Comment by sepalani** [comment]
Apparently, it does fix the issue: https://bugs.dolphin-emu.org/issues/12410#note-2

File: Source/Core/Core/Debugger/PPCDebugInterface.cpp (hunk 1 above)
Comment (leoetlino) [inline Source/Core/Core/Debugger/PPCDebugInterface.cpp:186]: Nitpick: 
```suggestion
  std::vector<u32> visited_addrs{{active_thread->GetAddress()}};
```

Comment (sepalani) [reply Source/Core/Core/Debugger/PPCDebugInterface.cpp]: Done.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**